string domain
string variable
---
string value
